### ISSUE TITLE
Build fails

### ISSUE THREAD

**Issue by GerardRodes** [sqlc-dev/sqlc#2591]
### Version

1.20.0

### What happened?

Build is failing, updating `pg_query_go` should fix it

Related: https://github.com/pganalyze/pg_query_go/pull/96

### Relevant log output

```shell
❯ go install github.com/sqlc-dev/sqlc/cmd/sqlc@latest  
# github.com/pganalyze/pg_query_go/v4/parser
src_port_snprintf.c:374:1: error: conflicting types for ‘strchrnul’; have ‘const char *(const char *, int)’
  374 | strchrnul(const char *s, int c)
      | ^~~~~~~~~
In file included from ../../.gvm/pkgsets/go1.21/global/pkg/mod/github.com/pganalyze/pg_query_go/v4@v4.2.1/parser/include/c.h:61,
                 from src_port_snprintf.c:62:
/usr/include/string.h:286:14: note: previous declaration of ‘strchrnul’ with type ‘char *(const char *, int)’
  286 | extern char *strchrnul (const char *__s, int __c)
      |              ^~~~~~~~~
cc1: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-deprecated-non-prototype’ may have been intended to silence earlier diagnostics
```

### What operating system are you using?

Linux

**Comment by mvdan** [comment]
Yep, see https://github.com/pganalyze/pg_query_go/issues/95. sqlc should bump the dependency and probably tag a release.

**Comment by blizzy78** [comment]
The relevant PR should be #2583.

**Comment by kyleconroy** [comment]
This has been fixed on main and will be included in sqlc 1.21, which will be released on 9/6
